src/components/UrqlQuery.re

```diff
@@ -1,70 +1,57 @@
-[@bs.module "urql"] external queryComponent: ReasonReact.reactClass = "Query";
-
+[@bs.deriving abstract]
 type queryRenderPropsJs('a) = {
-  .
-  "fetching": bool,
-  "data": Js.Nullable.t('a),
-  "error": Js.Nullable.t(UrqlCombinedError.t),
-  "executeQuery":
-    option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
+  fetching: bool,
+  [@bs.optional] data: 'a,
+  [@bs.optional] error: UrqlCombinedError.t,
+  executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult)
 };
 
 type queryRenderProps('a) = {
   fetching: bool,
   data: option('a),
   error: option(UrqlCombinedError.t),
   executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
-  response: UrqlTypes.response('a),
+  response: UrqlTypes.response('a)
 };
 
+module QueryJs = {
+  [@bs.module "urql"] [@react.component]
+  external make:
+    (~query: string, ~variables: Js.Json.t, ~requestPolicy: string, ~children: queryRenderPropsJs('a) => React.element) =>
+    React.element =
+    "Query";
+}
+
 let urqlDataToRecord = (result: queryRenderPropsJs('a)) => {
-  let data = result##data |> Js.Nullable.toOption;
-  let error = result##error |> Js.Nullable.toOption;
+  let data = result->dataGet;
```

Comment:
Since these can still potentially be `undefined` / `null`, I _think_ we still need to call `Js.Nullable.toOption` on them (i.e. for `data` and `error`). So I imagine it looking like:
```suggestion
  let data = result->dataGet->Js.Nullable.toOption;
  let error = result->errorGet->Js.Nullable.toOption;
```

Does that make sense?